### DISCUSSION TITLE
Is there a way to use orderBy with populate?

### DISCUSSION THREAD

**Discussion by alex996** [mikro-orm/mikro-orm#2610] [Q&A]
Say I have a query

```ts
const campaigns = orm.em.find(Campaign, { month: "01" }, ["surveys", "surveys.participants"]);
```

When I read `campaigns[0].surveys[0].participants` I get an array where objects seem to be ordered by `updatedAt` (the most recently updated appear last). I'd like to sort them in their original insertion order, i.e. by `id`.

Is there a way to specify `orderBy` on `surveys.participants`? Here's the SQL I got from debugging:

```sql
# Current
select "e0".* from "participant" as "e0" where "e0"."survey_id" in (1) order by "e0"."survey_id" asc

# Desired
select "e0".* from "participant" as "e0" where "e0"."survey_id" in (1) order by "e0"."survey_id" asc, "e0"."id" asc
```

I'd prefer to order at the SQL level, but if it's not possible, can be it done in the model class (perhaps with an on-populate hook)?

Thanks

**Comment by alex996** [comment]
As a workaround, I am using [`orderBy` in the relationship decorator](https://mikro-orm.io/docs/decorators#manytomany). It's not what I was looking for but it does the job.

**Comment by Baterka** [comment]
Is there a different way to do it? What if I want different order for different queries.

**Reply by B4nan** [reply]
see my answer below

**Comment by B4nan** [comment]
Ordering works similarly to filtering, so this should work:

```ts
const campaigns = await em.find(Campaign, { month: "01" }, {
  populate: ["surveys", "surveys.participants"],
  orderBy: { surveys: { id: 'asc' } },
});
```

This will result in ordering by the `surveys.id` on both queries, if you want to have it only on the second one, use separate `em.populate()` statementh that again has the `orderBy` parameter that works the same.
